ompi/mpi/c/isendrecv.c

```diff
@@ -0,0 +1,168 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2021 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2008 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2010-2012 Oak Ridge National Labs.  All rights reserved.
+ * Copyright (c) 2013      Los Alamos National Security, LLC.  All rights
+ *                         reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * Copyright (c) 2021      Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021      Triad National Security, LLC. All rights
+ *                         reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include "ompi_config.h"
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/communicator/communicator.h"
+#include "ompi/communicator/comm_request.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/mca/pml/pml.h"
+#include "ompi/request/request.h"
+#include "ompi/memchecker.h"
+#include "ompi/runtime/ompi_spc.h"
+
+#if OMPI_BUILD_MPI_PROFILING
+#if OPAL_HAVE_WEAK_SYMBOLS
+#pragma weak MPI_Isendrecv = PMPI_Isendrecv
+#endif
+#define MPI_Isendrecv PMPI_Isendrecv
+#endif
+
+static const char FUNC_NAME[] = "MPI_Isendrecv";
+
+struct ompi_isendrecv_context_t {
+    opal_object_t super;
+    int nreqs;
+    ompi_status_public_t statuses[2];
+};
+
+typedef struct ompi_isendrecv_context_t ompi_isendrecv_context_t;
+OBJ_CLASS_INSTANCE(ompi_isendrecv_context_t, opal_object_t, NULL, NULL);
+
+static int ompi_isendrecv_completer_func (ompi_comm_request_t *request)
+{
+    ompi_isendrecv_context_t *context =
+        (ompi_isendrecv_context_t *) request->context;
+
+    /*
+     * Copy the status from the receive side of the sendrecv request?
+     * But what if the send failed?
+     *
+     * Probably need to bring up in the MPI forum.  
+     */
+    
+    OMPI_COPY_STATUS(&request->super.req_status, 
```

Comment:
If we always copy the status [0] why do you need the additional statuses array ?